### PR TITLE
New feature: attendance sheet and bug fixes (Without archive related features)

### PR DESCRIPTION
**Pull request by EdenWuyifan**
Everything and resolves from #179 without archive related functions

1. Fix bugs noted by Hanna:
  - If you edit the About Me/Us sections, you must check box next to photo if you want to keep it after editing. Otherwise, it will disappear
  - When editing the About Me section on volunteer profile, you must add badges. Just write None for now
  - The profile photo disappears when you go to My Voluncheers
2. Add a list of Volunteers who signed up for the event.
3. Add attendance sheet for organizations to track.
4. Link the attendance to the hours_volunteered field for badge feature.

image

### CODE REVIEW

File: voluncheer/profiles/views/profile.py [hunk 1]
```diff
@@ -97,26 +100,59 @@ def password_reset_request(request):
...
+    """
+    confirm_attendees = request.GET.getlist("volunteer-attended")
+    opportunity = get_object_or_404(Opportunity, pk=opportunity_id)
+    for attendee_pk in confirm_attendees:
+        volunteer = Volunteer.objects.get(pk=attendee_pk)
+        if volunteer not in opportunity.volunteers.all():
+            continue
+        if volunteer in opportunity.attended_volunteers.all():
+            opportunity.attended_volunteers.remove(volunteer)
+            volunteer.hours_volunteered -= opportunity.duration
```
Comment (Hgersten-hash) [inline voluncheer/profiles/views/profile.py:151]: Why are we subtracting the hours here from the Volunteer? If they didn't attend, then they just don't get the hours added, right?

File: voluncheer/templates/profiles/profile.html [hunk 2]
```diff
@@ -74,7 +73,28 @@ <h2>POSTED OPPORTUNITIES</h2>
...
+                        <form method="get" id="opportunities-detail-{{opportunity.pk}}-form" class="opportunities-detail-form" action="{% url 'attendance' opportunity.pk %}">
+                            <div id="volunteer-attandence">
+                                <p class="vol-position-disciption-h1">Volunteer attendance:</p>
+                                {% for volunteer in opportunity.volunteers.all %}
+                                    {% if volunteer in opportunity.attended_volunteers.all %}
+                                    <div class="sticky-note green">
+                                        <input type="checkbox" class="volunteer-attandence" name="volunteer-attended" value="{{volunteer.pk}}">Attended
+                                    {% else %}
+                                    <div class="sticky-note">
+                                        <input type="checkbox" class="volunteer-attandence" name="volunteer-attended" value="{{volunteer.pk}}">Not Attend
```
Comment (Hgersten-hash) [inline voluncheer/templates/profiles/profile.html:85]: Did Not Attend instead of Not Attend

File: voluncheer/templates/profiles/profile.html [hunk 3]
```diff
@@ -74,7 +73,28 @@ <h2>POSTED OPPORTUNITIES</h2>
...
+                                    {% endif %} 
+                                        <br>
+                                        Name: {{volunteer.first_name}}, {{volunteer.last_name}}
+                                        {% if volunteer.photo %}
+                                        <img src="{{volunteer.photo.url}}" alt="profile pic" width="100" height="100"/>
+                                        {% endif %}
+                                    </div>
+                                {% endfor %}
+                            </div>
+                            <button type="submit" id="opportunities-detail-form-btn" class="opportunities-detail-form-submit">Confirm Attandence</button>
```
Comment (Hgersten-hash) [inline voluncheer/templates/profiles/profile.html:95]: Confirm Attendance

**Review by Hgersten-hash — CHANGES_REQUESTED** [review]

File: voluncheer/profiles/views/profile.py (hunk 1 above)
Comment (EdenWuyifan) [reply voluncheer/profiles/views/profile.py:151]: This function will be triggered both for confirm attendance and cancel attendance, click again will cancel their attendance.

File: voluncheer/templates/profiles/profile.html (hunk 3 above)
Comment (EdenWuyifan) [reply voluncheer/templates/profiles/profile.html]: Done

File: voluncheer/profiles/views/profile.py (hunk 1 above)
Comment (Hgersten-hash) [reply voluncheer/profiles/views/profile.py:151]: every time the function is triggered, will it subtract the hours from a volunteer who didn't attend?

Comment (Hgersten-hash) [reply voluncheer/profiles/views/profile.py:151]: Oh, I think I see now. The function checks to see if the volunteer is in attended_volunteers first, and if they are and we are canceling their attendance then they are removed?

Comment (EdenWuyifan) [reply voluncheer/profiles/views/profile.py:151]: Yeah! That's correct!

Comment (Hgersten-hash) [reply voluncheer/profiles/views/profile.py:153]: Got it! Just to confirm, in order to set a volunteer as not attended, the organization would have to click on the checkbox again?

Comment (EdenWuyifan) [reply voluncheer/profiles/views/profile.py:153]: yeah, thats right

**Pull request merged**